src/embind/embind_ts.js

```diff
@@ -0,0 +1,276 @@
+    registerPrimitiveType(rawType, name);
+  },
+  _embind_register_memory_view: function(rawType, dataTypeIndex, name) {
+    // TODO
+  },
+  _embind_register_function__deps: ['$moduleDefinitions', '$createFunctionDefinition'],
+  _embind_register_function: function(name, argCount, rawArgTypesAddr, signature, rawInvoker, fn, isAsync) {
+    createFunctionDefinition(name, argCount, rawArgTypesAddr, false, (funcDef) => {
+      moduleDefinitions.push(funcDef);
+    });
+  },
+  _embind_register_class__deps: ['$readLatin1String', '$ClassDefinition', '$whenDependentTypesAreResolved', '$moduleDefinitions'],
+  _embind_register_class: function(rawType,
+                                  rawPointerType,
+                                  rawConstPointerType,
+                                  baseClassRawType,
+                                  getActualTypeSignature,
+                                  getActualType,
+                                  upcastSignature,
+                                  upcast,
+                                  downcastSignature,
+                                  downcast,
+                                  name,
+                                  destructorSignature,
+                                  rawDestructor) {
+    name = readLatin1String(name);
+    whenDependentTypesAreResolved(
+      [rawType, rawPointerType, rawConstPointerType],
+      baseClassRawType ? [baseClassRawType] : [],
+      function(base) {
+        var classDef = new ClassDefinition(rawType, name, base.length ? base[0] : null);
+        moduleDefinitions.push(classDef);
+        return [classDef, classDef, classDef];
+      }
+    );
+
+  },
+  _embind_register_class_constructor__deps: ['$whenDependentTypesAreResolved'],
+  _embind_register_class_constructor: function(
+    rawClassType,
+    argCount,
+    rawArgTypesAddr,
+    invokerSignature,
+    invoker,
+    rawConstructor
+  ) {
+    whenDependentTypesAreResolved([], [rawClassType], function(classType) {
+      classType = classType[0];
+      createFunctionDefinition(`constructor ${classType.name}`, argCount, rawArgTypesAddr, false, (funcDef) => {
+        classType.constructors.push(funcDef);
+      });
+      return [];
+    });
+  },
+  _embind_register_class_function__deps: ['$createFunctionDefinition'],
+  _embind_register_class_function: function(rawClassType,
+          methodName,
+          argCount,
+          rawArgTypesAddr, // [ReturnType, ThisType, Args...]
+          invokerSignature,
+          rawInvoker,
+          context,
+          isPureVirtual,
+          isAsync) {
+    createFunctionDefinition(methodName, argCount, rawArgTypesAddr, true, (funcDef) => {
+      let classDef = funcDef.thisType;
+      classDef.methods.push(funcDef);
+    });
+  },
+
+  _embind_initialize_done__deps: ['$awaitingDependencies', '$throwBindingError', '$getTypeName', '$moduleDefinitions', '$TsPrinter'],
+  _embind_initialize_done: function() {
```

Comment:
If the embind initialization happens as a static constructor I think we can instead try to hook into an existing mechanism for running code after that point.

I think the should be able to just do something like `addOnInit(embindEmitTypes)` in the `__postset` for `$embindEmitTypes` and then add `$embindEmitTypes` to `DEFAULT_LIBRARY_FUNCS_TO_INCLUDE`.

Then you should be able to avoid the new native symbol.